manifests/00-crd.yaml

```diff
@@ -209,7 +620,27 @@ spec:
                         privateKey:
                           description: PrivateKey points to secret containing the
                             private key, provided by AWS.
+                          properties:
+                            key:
+                              description: The key of the secret to select from.  Must
+                                be a valid secret key.
+                              type: string
+                            name:
+                              description: 'Name of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
+                                TODO: Add other useful fields. apiVersion, kind, uid?'
+                              type: string
+                            optional:
+                              description: Specify whether the Secret or it's key
+                                must be defined
+                              type: boolean
+                          required:
+                          - key
                           type: object
+                      required:
+                      - baseURL
+                      - duration
+                      - keypairID
+                      - privateKey
```

Comment:
Looks like our CRD generator is interpreting everything as required. API needs to be updated to have `// +optional` in the doc comment and `omitempty` in the json tag.

Refer to upstream api doc, types may also need to be switched to pointers - https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required

Reply:
`// +optional` is enough. I added this comment tag to the optional fields.